### PR TITLE
Fixes #14596: Add a GM to verify that a given key-value  in a file matches a regex

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14596

### CODE REVIEW

File: tree/30_generic_methods/file_key_value_match_regex.cf [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File key-value match regex
+# @description Ensure that one parameter matches a given regex, on one single line, in the right hand side of a key->values line
+# @documentation Audit the file, and verify that the `values` part from a key->values line matches a given regex.
+#
+# #### Example
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_match_regex.cf]: Wrong example

File: tree/30_generic_methods/file_key_value_match_regex.cf [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  vars:
+      "old_class_prefix"  string => canonify("file_key_value_match_regex_${file}");
+      "args"               slist => { "${file}", "${key}", "${key_value_separator}", "${regex}" };
+      "report_param"      string => join("_", args);
+      "full_class_prefix" string => canonify("file_key_value_match_regex_${report_param}");
+      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
+
+
+      "escaped_key"       string => escape("${key}");
+      "escaped_parameter" string => escape("${parameter}");
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_match_regex.cf]: unused

File: tree/30_generic_methods/file_key_value_match_regex.cf [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      "success class" usebundle  => _classes_success("${class_prefix}");
+    !keys_found|failed::
+      "error_class"   usebundle => _classes_failure("${old_class_prefix}");
+      "error_class"   usebundle => _classes_failure("${class_prefix}");
+    pass2.keys_found::
+      "report"   usebundle => _log_v3("${report_string}Verify that parameters in lines ${key} ${key_value_separator} values in ${file} match the regx ${regex}", "${file}", "${old_class_prefix}", "${class_prefix}", @{args});
+    pass2.!keys_found::
+      "report"   usebundle => _log_v3("No lines found corresponding to the given key, verify that parameters in lines ${key} ${key_value_separator} values in ${file} match the regx ${regex}", "${file}", "${old_class_prefix}", "${class_prefix}", @{args});
+
+  reports:
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_match_regex.cf]: to remove

**Comment by peckpeck** [comment]
As we said before, this should not be merged in a release branch

File: tree/30_generic_methods/file_key_value_match_regex.cf [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      "full_class_prefix" string => canonify("file_key_value_match_regex_${report_param}");
+      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
+
+
+      "escaped_key"       string => escape("${key}");
+      "escaped_parameter" string => escape("${parameter}");
+
+      "lines_list"  slist => readstringlist("${file}", "", "\n", "99999999999", "99999999999");
+      "lines_length"  int => length("lines_list");
+      "indices" slist => expandrange("[1-${lines_length}]", 1);
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_match_regex.cf]: 0-${lines_length}

**Comment by Fdall** [comment]
Not relevant anymore. We will split the parsing part and the checking one.

**Pull request closed without merging**